Source/WebKit/UIProcess/RemoteLayerTree/mac/RemoteLayerTreeEventDispatcher.cpp

```diff
@@ -101,7 +101,14 @@ RemoteLayerTreeEventDispatcher::RemoteLayerTreeEventDispatcher(RemoteScrollingCo
 {
 }
 
-RemoteLayerTreeEventDispatcher::~RemoteLayerTreeEventDispatcher() = default;
+RemoteLayerTreeEventDispatcher::~RemoteLayerTreeEventDispatcher()
+{
+    m_beingDestroyed = true;
```

Comment:
Should not we make `MomentumEventDispatcher::m_client` a pointer and add the function `MomentumEventDispatcher::clearClient()` and call it here: `m_momentumEventDispatcher->clearClient();`?